### DISCUSSION TITLE
Strange parttern of temporalCoherence

### DISCUSSION THREAD

**Discussion by comma-never-coma** [insarlab/MintPy#1013] [Q&A] [answered]
Hi everyone!
I am dealing with a long time series(2020-2023) aquired from ISCE topsStack, and smallbaselineApp.py has finished running successfully. However, a large segment of the deformation result is masked due to a minTempCoh threshold. 
The study area is divided into upper and lower parts by rivers, as shown in the figure. The `spatial coherence` seems normal, but the `temporal coherence` of the lower part is very low. Similarly, the value of `numTriNonzeroInt` is larger in the lower part of the region. I'm not sure why there is such a significant difference between the two, could this possibly be caused by SNAPHU phase unwrapping in the two separate regions?
Any references or advice is appreciated!
![avgSpatialCoh](https://github.com/insarlab/MintPy/assets/67263104/5f004b3e-f06d-4fdf-b906-66150304072c)
![temporalCoherence](https://github.com/insarlab/MintPy/assets/67263104/55c17ea5-5c1e-4703-99f6-bddb355c35f7)
![numTriNonzeroIntAmbiguity](https://github.com/insarlab/MintPy/assets/67263104/f44deced-39f8-4bb9-8f19-44f3bee37e93)

**Comment by yunjunz** [comment, marked as answer]
This is a classic example of phase unwrapping errors, in the delta and southern side of the river. You could find similar patterns in the San Francisco dataset from ARIA in previous years' UNAVCO short course, or in Yunjun et al (2019) section 3. It could be corrected using the unwrapping error corrections method in mintpy.

**Reply by comma-never-coma** [reply]
Thanks for your explanation! By the way, regarding the `connCompMinArea` parameter, the default value in the template file is 2500. When the size of my interferograms is 1890*1399, running the unwrapping error correction module directly would cause an error. The explanation of the `connCompMinArea` is "discard regions smaller than the min size in pixels". I guess the parameter value should be less than the interferogram's dimensions, but I am uncertain about the exact suitable value for it.

**Reply by yunjunz** [reply]
2500 pixels means 50x50 for example, in the connected components, based on which all PU error corrections methods are applied. You could try different values to test in your own cases.

**Reply by comma-never-coma** [reply]
I got it. Thank you so much!
